expertise/service/expertise.py

```diff
@@ -79,11 +80,13 @@ def _prepare_config(self, request) -> dict:
                         when it is not expected
         """
         # Validate fields
-        config = self._validate_fields(request)
-        self.logger.info(f"Config validation passed - setting server-side fields")
+        validated_request = APIRequest()
+        validated_request.from_json(request)
+        config = ServerConfig(self._get_default_config(), validated_request).to_json()
+        self.logger.info(f"Config validation passed - setting server-side fields in {config}")
 
         # Populate with server-side fields
-        root_dir = os.path.join(self.working_dir, request['job_id'])
+        root_dir = os.path.join(self.working_dir, config['job_id'])
```

Comment:
if config is an instance of `ServerConfig` , shouldn't be `config.job_id` here?

Reply:
`config` gets the JSON value of `ServerConfig` because the expertise model only accepts a dict. I could for example, change the model so that it can load from the fields of a `ServerConfig` (`JobConfig` from your later comment) and that would let us use a `JobConfig` object throughout for consistency.

Reply:
Now, I understand why you can do this, you are calling to_json after creating the JobConfig object, why?

Can't you just use the config object? then when you need to run the model you can save the config object to a json file.

Reply:
you can also move this code to the JobConfig class:

```
        # Set SPECTER+MFR paths
        if 'specter' in config.get('model', 'specter+mfr'):
            config['model_params']['specter_dir'] = self.specter_dir
        if 'mfr' in config.get('model', 'specter+mfr'):
            config['model_params']['mfr_feature_vocab_file'] = self.mfr_feature_vocab_file
            config['model_params']['mfr_checkpoint_dir'] = self.mfr_checkpoint_dir

```